tasks/orca-recovery-adapter/src/index.ts

```diff
@@ -0,0 +1,127 @@
+import { v4 as uuidv4 } from 'uuid';
+import { Context } from 'aws-lambda';
+
+import { sfn } from '@cumulus/aws-client/services';
+import { describeExecution } from '@cumulus/aws-client/StepFunctions';
+import { runCumulusTask, CumulusMessageWithAssignedPayload } from '@cumulus/cumulus-message-adapter-js';
+import Logger from '@cumulus/logger';
+import { buildExecutionArn } from '@cumulus/message/Executions';
+import { CumulusMessage, CumulusRemoteMessage } from '@cumulus/types/message';
+
+import { HandlerEvent, HandlerOutput } from './types';
+
+const log = new Logger({ sender: '@cumulus/orca-recovery-adapter' });
+
+/**
+ * Get state machine execution results
+ *
+ * @param {object} params
+ * @param {string} params.executionArn - execution arn
+ * @param {number} params.retries - number of retries
+ * @param {number} params.retryIntervalInSecond - retry internal in second
+ * @param {number} params.maxRetryTimeInSecond - max retry time in second
+ * @returns Promise<AWS.StepFunctions.DescribeExecutionOutput> - Returns promise
+ *   that resolves to the output of step function execution
+ */
+export const getStateMachineExecutionResults = async (
+  params: {
+    executionArn: string,
+    retries?: number,
+    retryIntervalInSecond?: number,
+    maxRetryTimeInSecond?: number,
+  }
+) : Promise<AWS.StepFunctions.DescribeExecutionOutput> => {
+  const {
+    executionArn,
+    retries = 50,
+    retryIntervalInSecond = 5,
+    maxRetryTimeInSecond = 1800,
+  } = params;
+  const result = await pRetry(
+    async () => {
+      const response = await describeExecution({ executionArn });
+      if (response.status === 'RUNNING') {
+        throw new Error(`Waiting for recovery workflow ${executionArn} to complete`);
+      }
+      return response;
+    },
+    {
+      retries,
+      minTimeout: retryIntervalInSecond * 1000,
+      maxTimeout: maxRetryTimeInSecond * 1000,
+      maxRetryTime: maxRetryTimeInSecond * 1000,
+      onFailedAttempt: (error) => {
+        log.debug(`Attempt ${error.attemptNumber} times to get result ${executionArn}. ${error.retriesLeft} remain at ${new Date().toString()}`);
+      },
+    }
+  );
+  return result;
+};
+
+/**
+ * Invoke ORCA Recovery workflow
+ *
+ * @param {HandlerEvent} event - input from the message adapter
+ * @returns {Promise<HandlerOutput>} - returns output from ORCA workflow
+ */
+export const invokeOrcaRecoveryWorkflow = async (
+  event: HandlerEvent
+) : Promise<HandlerOutput> => {
+  const workflowArn = process.env.orca_sfn_recovery_workflow_arn;
+  if (!workflowArn?.length) {
+    log.error('Environment orca_sfn_recovery_workflow_arn is not set');
+    throw new Error('Environment orca_sfn_recovery_workflow_arn is not set');
+  }
+
+  const payload = {
+    payload: event.input,
+    config: event.config,
+  };
+
+  const currentWorkflowArn = buildExecutionArn(
+    event.cumulus_config?.state_machine || '',
+    event.cumulus_config?.execution_name || ''
```

Comment:
```suggestion
    event.cumulus_config?.state_machine!,
    event.cumulus_config?.execution_name!
```

should we assert these aren't undefined? Feels like it'll produce a broken arn if they are empty string.

Reply:
I added empty string to avoid typescript error
`Argument of type 'string | undefined' is not assignable to parameter of type 'string'.`
It's a good idea to use non-null assertion operator.  Thanks!